### ISSUE TITLE
Property does not exist: $types

### ISSUE THREAD

**Issue by far-blue** [auraphp/Aura.Di#154]
Hi,

I'm getting the following error:
`Uncaught Aura\Di\Exception\NoSuchProperty: Property does not exist: $types`

when trying to do this:
```
class DiConfig extends ContainerConfig
{
	public function define(Container $c)
	{
		$c->types[Connection\ConnectionInterface::class] = $c->lazyNew(Connection\AmqpConnection::class);
	}
}
```

I have similar setup in other projects which works but where I'm not using ContainerConfig.

Any ideas?

**Comment by far-blue** [comment]
Ok, I've worked it out. `Resolver` doesn't do `types[]`, `AutoResolver` does. By default, `ContainerBuilder` uses `Resolver` and you need to pass true to the second parameter of `newConfiguredInstance()` to use `AutoResolver`.

**Comment by far-blue** [comment]
Maybe this is worth mentioning on this page in the docs: http://auraphp.com/packages/3.x/Di/config.html#1-1-8

**Comment by pmjones** [comment]
It *is* worth mentioning. You feel like sending along a PR for that doc page?

**Comment by harikt** [comment]
From the docs : http://auraphp.com/packages/3.x/Di/auto.html#1-1-9-1 , I feel the types usage is already mentioned to use `$builder::AUTO_RESOLVE` . Or did you came across the types some where else ?

**Comment by far-blue** [comment]
Maybe I'm unusual in doing so but I tend to read docs with an intention rather than page-by-page. I wanted to get started so read the Getting Started page which mentioned a ContainerBuilder so I jumped to that section and created a config class. Then I read about get and newInstance and figured get() was the best bet for my bootstrap to get everything going so read about Services. One of my parameters was an interface so I saw section 1.1.9.2 in the contents page and jumped to it and tried to follow the example. As such, I missed the stuff at the top of section 1.1.9. I think mentioning it at the point in the docs where you are learning about ContainerBuilder would help people like me :) I'll sort out a PR for discussion :)
